### PR TITLE
fix: specify content type to application/json when it throws an invalid graphql request error

### LINKED ISSUE #2320: Invalid content type on invalid graphql request errors
**Describe the bug**
If an invalid GraphQL request is sent to the router, router does not set `content-type: application/json` even though the response is JSON, and looks like this:

```json
{
	"message": "Invalid GraphQL request",
	"extensions": {
		"details": "failed to deserialize the request body into JSON: EOF while parsing a string at line 1 column 11",
		"code": "INVALID_GRAPHQL_REQUEST"
	}
}
```

P.S. The new error handling done in 1.7.0 is great, just noticed this slight regression.

**To Reproduce**
1. Start router with any schema
2. Do a curl request with invalid payload, e.g `-H 'content-type: application/json' -d '{"query":"}'`

**Expected behavior**
Response should have `content-type: application/json` header

### PR DESCRIPTION
**Pull request by bnjjj**
close #2320

### CODE REVIEW

File: NEXT_CHANGELOG.md [hunk 1]
```diff
@@ -32,4 +32,12 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
...
 
-By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2310
\ No newline at end of file
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2310
+
+## 🛠 Maintenance
+
+### Specify content type to `application/json` when it throws an invalid GraphQL request error ([Issue #2320](https://github.com/apollographql/router/issues/2320))
+
+When throwing a `INVALID_GRAPHQL_REQUEST` error, it nows specify the right content-type header.
```
Comment (abernix) [inline NEXT_CHANGELOG.md:41]:
```suggestion
When throwing a `INVALID_GRAPHQL_REQUEST` error, it nows specify the right `content-type` header.
```

**Review by abernix — CHANGES_REQUESTED** [review]
This looks great, but needs a test before we land it — both for the `INVALID_GRAPHQL_REQUEST` response itself (of which there is currently none) and for the presence of the `content-type: application/json` header being present in that response.

**Comment by bnjjj** [comment]
test added @abernix

**Review by abernix — APPROVED** [review]
nice!

**Review by o0Ignition0o — APPROVED** [review]

File: NEXT_CHANGELOG.md [hunk 2]
```diff
@@ -32,4 +32,12 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
...
 
-By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2310
\ No newline at end of file
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2310
+
+## 🛠 Maintenance
+
+### Specify content type to `application/json` when it throws an invalid GraphQL request error ([Issue #2320](https://github.com/apollographql/router/issues/2320))
+
+When throwing a `INVALID_GRAPHQL_REQUEST` error, it nows specify the right `content-type` header.
```
Comment (Geal) [inline NEXT_CHANGELOG.md:41]:
```suggestion
When throwing a `INVALID_GRAPHQL_REQUEST` error, it now specifies the right `content-type` header.
```

**Review by Geal — APPROVED** [review]

**Pull request merged**